### PR TITLE
CATALOG-9786: [update] Brands: REST Catalog

### PR DESCRIPTION
**Pull request by NataliiaNefodova**
# [CATALOG-9786]

## What changed?
* Add swagger file for V2 Brands
* Update swagger file for V3 Brands

## Release notes draft
* Revert this PR

[CATALOG-9786]: https://bigcommercecloud.atlassian.net/browse/CATALOG-9786?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by olysenko — DISMISSED** [review]

File: archive/rest-catalog/brands.v2.yaml [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+        The following properties of the brand are required. The request won’t be fulfilled unless these properties are valid.
+        - name
+        
+        **Read-only Fields**
+        
+        The following properties of the brand are read-only. If one or more of these properties are included in the request, it will be rejected.
+        - id
+        
+        **Limits**
+        - 30 000 brands per store limit
```
Comment (bc-traciporter) [inline archive/rest-catalog/brands.v2.yaml:69]:
```suggestion
        - 30,000 brands per store limit
```

File: archive/rest-catalog/brands.v2.yaml [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+        To maximize system performance, BigCommerce caps the number of brands that can be added to a store at 30,000. If your POST causes the store to exceed the maximum of 30,000 brands, BigCommerce will return a 403 error.
```
Comment (bc-traciporter) [inline archive/rest-catalog/brands.v2.yaml:70]:
```suggestion
        To maximize system performance, BigCommerce caps the number of brands you can add to a store at 30,000. If your POST causes the store to exceed the maximum of 30,000 brands, BigCommerce will return a 403 error.
```

File: archive/rest-catalog/brands.v2.yaml [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+        '401':
+          $ref: '#/components/responses/errorsWithDetails'
+        '403':
+          $ref: '#/components/responses/errorsWithDetails'
+        '409':
+          $ref: '#/components/responses/errorsWithDetails'
+    delete:
+      tags:
+        - Brands V2
+      summary: Deletes all existing on store brands.
```
Comment (bc-traciporter) [inline archive/rest-catalog/brands.v2.yaml:95]:
```suggestion
      summary: Deletes all existing on store brands. (Deprecated)
```

File: archive/rest-catalog/brands.v2.yaml [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+          content: { }
+        '401':
+          $ref: '#/components/responses/UnauthorizedError'
+        '403':
+          $ref: '#/components/responses/errorsWithDetails'
+  '/brands/{brand_id}':
+    get:
+      tags:
+        - Brands V2
+      summary: Gets a brand.
```
Comment (bc-traciporter) [inline archive/rest-catalog/brands.v2.yaml:110]:
```suggestion
      summary: Gets a brand. (Deprecated)
```

File: archive/rest-catalog/brands.v2.yaml [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+        '401':
+          $ref: '#/components/responses/errorsWithDetails'
+        '403':
+          $ref: '#/components/responses/errorsWithDetails'
+        '404':
+          $ref: '#/components/responses/NotFoundError'
+    put:
+      tags:
+        - Brands V2
+      summary: Updates an existing Brand.
```
Comment (bc-traciporter) [inline archive/rest-catalog/brands.v2.yaml:127]:
```suggestion
      summary: Updates an existing Brand. (Deprecated)
```

File: archive/rest-catalog/brands.v2.yaml [hunk 6]
```diff
@@ -0,0 +1,515 @@
...
+        '401':
+          $ref: '#/components/responses/errorsWithDetails'
+        '403':
+          $ref: '#/components/responses/errorsWithDetails'
+        '404':
+          $ref: '#/components/responses/NotFoundError'
+    delete:
+      tags:
+        - Brands V2
+      summary: Deletes a Brand.
```
Comment (bc-traciporter) [inline archive/rest-catalog/brands.v2.yaml:158]:
```suggestion
      summary: Deletes a Brand. (Deprecated)
```

File: archive/rest-catalog/brands.v2.yaml [hunk 7]
```diff
@@ -0,0 +1,515 @@
...
+          $ref: '#/components/responses/errorsWithDetails'
+        '404':
+          $ref: '#/components/responses/NotFoundError'
+    parameters:
+      - $ref: '#/components/parameters/BrandIdParam'
+  '/brands/count':
+    get:
+      tags:
+        - Brands V2
+      summary: Count Brands.
```
Comment (bc-traciporter) [inline archive/rest-catalog/brands.v2.yaml:177]:
```suggestion
      summary: Count Brands. (Deprecated)
```

**Review by bc-traciporter — COMMENTED** [review]
@NataliiaNefodova I left comments.

File: archive/rest-catalog/brands.v2.yaml (hunk 1 above)
Comment (NataliiaNefodova) [reply archive/rest-catalog/brands.v2.yaml:69]: Changed

File: archive/rest-catalog/brands.v2.yaml (hunk 2 above)
Comment (NataliiaNefodova) [reply archive/rest-catalog/brands.v2.yaml:70]: Changed. Thanks!

File: archive/rest-catalog/brands.v2.yaml [hunk 8]
```diff
@@ -0,0 +1,515 @@
...
+        '400':
+          $ref: '#/components/responses/errors'
+        '401':
+          $ref: '#/components/responses/errorsWithDetails'
+        '403':
+          $ref: '#/components/responses/errorsWithDetails'
+    post:
+      tags:
+        - Brands V2
+      summary: Create a Brand.
```
[the remark this answers is not in the document]
Comment (NataliiaNefodova) [reply archive/rest-catalog/brands.v2.yaml]: Changed to "summary: Create a Brand. (Deprecated)"

File: archive/rest-catalog/brands.v2.yaml (hunk 4 above)
Comment (NataliiaNefodova) [reply archive/rest-catalog/brands.v2.yaml]: Changed to "summary: Gets a brand. (Deprecated)"

File: archive/rest-catalog/brands.v2.yaml (hunk 6 above)
Comment (NataliiaNefodova) [reply archive/rest-catalog/brands.v2.yaml]: Changed to "summary: Deletes a Brand. (Deprecated)"

File: archive/rest-catalog/brands.v2.yaml (hunk 7 above)
Comment (NataliiaNefodova) [reply archive/rest-catalog/brands.v2.yaml]: Changed to "summary: Count Brands. (Deprecated)"

File: archive/rest-catalog/brands.v2.yaml [hunk 9]
```diff
@@ -0,0 +1,515 @@
...
+security:
+  - basicAuth: []
+tags:
+  - name: Brands V2
+paths:
+  '/brands':
+    get:
+      tags:
+        - Brands V2
+      summary: Gets list of Brands.
```
[the remark this answers is not in the document]
Comment (NataliiaNefodova) [reply archive/rest-catalog/brands.v2.yaml]: Changed to "summary: Gets list of Brands. (Deprecated)"

File: archive/rest-catalog/brands.v2.yaml (hunk 3 above)
Comment (NataliiaNefodova) [reply archive/rest-catalog/brands.v2.yaml]: Changed to "summary: Deletes all existing on store brands. (Deprecated)"

File: archive/rest-catalog/brands.v2.yaml (hunk 5 above)
Comment (NataliiaNefodova) [reply archive/rest-catalog/brands.v2.yaml]: Changed to "summary: Updates an existing Brand. (Deprecated)"

**Review by bc-tgomez — APPROVED** [review]
Looks great!

**Pull request merged**